Backport Cisco ACI CNI changes for OpenShift Version 4.17 on OpenStack

abhis2112 opened the pull request.

PariyaAshok committed:
add cisco aci cni security group rule security-groups.yaml.

(cherry picked from commit 87f0ac708d97d10168c849b24e2d7502e326c151)

PariyaAshok committed:
add cisco aci cni changes in network.yaml

(cherry picked from commit d18ad8d0b0fb56d897738b652db74a314fb61a81)

PariyaAshok committed:
add playbook to create ACI containers node network

(cherry picked from commit 24299834942b817b40b25e59a15ff0c20f280708)

PariyaAshok committed:
add cisco aci cni changes in control-plane.yaml

(cherry picked from commit 8b7f4cbec6f5bcd2db244353ee0be13eb373a160)

PariyaAshok committed:
add playbook to disable octavia lb service

(cherry picked from commit b3398481ea793666a2c9b4cd6b6c33d139ead9d9)

PariyaAshok committed:
add network scripts for control and compute nodes to bootstrap.ign, also template to create snatpolicy

(cherry picked from commit d9d40b8bba04a940b9ce7cc034baa702146feb57)

PariyaAshok committed:
add playbook to create snatpolicy

(cherry picked from commit d54cb8d264e6e6d193db09b257ca4f56c8d90d42)

PariyaAshok committed:
add post install script.

(cherry picked from commit 27d475100532e727d5adcd653d604173e8fe7e69)

PariyaAshok committed:
add common_ciscoaci.yaml

(cherry picked from commit 4783e17670183cf1f105edba0f370a9a45a1a323)

PariyaAshok committed:
add cisco aci cni changes for delete workflow

(cherry picked from commit 40209eb6a149fcea736bd4508e064dad1067f7a0)

PariyaAshok committed:
update inventory.yaml with cisco aci cni specific configuration fields

(cherry picked from commit 75f3e99852a99a4b605f4d0c9d2da5da84978a8b)

PariyaAshok committed:
add back upstream changes to create IPV6 subnet

(cherry picked from commit 5bad89baa4af666de76c606d6f9722d6fb94f860)
(cherry picked from commit 6b669f7736b165f55ee690cb65dac2d4ed0e1803)

PariyaAshok committed:
done some minor code cleanup

(cherry picked from commit 4a5dff24bea431610c44c721bf868c98bba5185e)
(cherry picked from commit fca7df63d1bb1f6f0ab380c4d5e4aa15c818fd07)

maheshkurund commented on upi/openstack/network.yaml:
```diff
@@ -34,6 +76,86 @@
+      - os_networking_type == "CiscoACI"
+      - primary_cluster_subnet.stdout == ""
+
+  - name: 'Create the cluster IPv6 subnet'
```
This entry to create IPv6 subnet should be removed as we already defined above.

abhis2112 replied:
Thanks @maheshkurund I have rectified it.
For security group issue - we have to remove it from kmr2-release-4.16 as well as i can see same changes on 4.16 1. https://github.com/noironetworks/openshift_installer/pull/203/commits/ad3dcb5411557b1825ba27573687c91549a387eb#diff-810ef9226dc20e51f9b2814e9885d243be1590bff38badb8d2b341263915d30cL194-L195
2. https://github.com/noironetworks/openshift_installer/pull/203/commits/ad3dcb5411557b1825ba27573687c91549a387eb#diff-810ef9226dc20e51f9b2814e9885d243be1590bff38badb8d2b341263915d30cL226-L227

abhis2112 committed:
Refactor network.yaml based on 4.17 changes
1. Remove "Create the cluster IPv6 subnet" wth os_subnet6 field
2. Remove security_groups addition during api/ingress port creation.
   This is being handled in update-network-resources.yaml
